compiler-rt/lib/sanitizer_common/sanitizer_stoptheworld_linux_libcdep.cpp

```diff
@@ -405,17 +405,20 @@ struct ScopedSetTracerPID {
 
 // This detects whether ptrace is blocked (e.g., by seccomp), by forking and
 // then attempting ptrace.
-// This separate check is necessary because StopTheWorld() creates a *thread*,
-// and therefore cannot use waitpid() due to the shared errno.
+// This separate check is necessary because StopTheWorld() creates a child
+// process with a shared virtual address space, and therefore cannot use
```

Comment:
I don't think this is true though. Thread locals should be relative to the thread pointer, which should be different. That wasn't a statement on my part, but a question (you can do all sorts of messed up stuff with clone flags)